projects/rero/ng-core/src/lib/record/search/record-search.component.ts

```diff
@@ -545,13 +545,19 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
 
   /**
    * Get export format url.
-   * @param format - string, export format
+   * @param format - export format object
    * @return formatted url for an export format.
    */
-  getExportFormatUrl(format: string) {
-    const baseUrl = this._apiService.getEndpointByType(this._currentIndex());
-    let url = `${baseUrl}?q=${encodeURIComponent(this.q)}&format=${format}&size=${RecordService.MAX_REST_RESULTS_SIZE}`;
+  getExportFormatUrl(format: any) {
+    const baseUrl = format.endpoint
+      ? format.endpoint
+      : this._apiService.getEndpointByType(this._currentIndex());
+    let url = `${baseUrl}?q=${encodeURIComponent(this.q)}&format=${format.format}`;
 
+    // check if max rest result size is disabled
+    if (!format.disableMaxRestResultsSize) {
+      url += `&size=${RecordService.MAX_REST_RESULTS_SIZE}`;
+    }
```

Comment:
query string creation seems a little bit confused.
If you have courage and determination, it exists a pretty solution using `Router` and `URLSerializer` : https://stackoverflow.com/a/51025291

Reply:
To be tested when we have more time. I mentionned that in a TODO